### DISCUSSION TITLE
File/Prompt/Instruction input for server

### DISCUSSION THREAD

**Discussion by ASH1998** [ggml-org/llama.cpp#2671] [Q&A] [answered]
Is there a way to send instructions to the server to give output in one type of format ?              
There is a syntax in ./main `-p` where you give instructions, I am looking for something similar on `./server`

In the server UI, there is a prompt box, it works similarly, any input is appreciated, thanks.

![Screenshot 2023-08-19 at 10 01 09 PM](https://github.com/ggerganov/llama.cpp/assets/23384411/e54610aa-ac1e-4f3e-a8ca-f4b2d954f543)

**Comment by ianscrivener** [comment]
Sounds like might be looking for the **grammer** functionality - eg output the response as JSON. [#2494](https://github.com/ggerganov/llama.cpp/discussions/2494), [#2560](https://github.com/ggerganov/llama.cpp/discussions/2560), [#1773](https://github.com/ggerganov/llama.cpp/pull/1773)

Also worth reviewing: **system prompt** - [#2637](https://github.com/ggerganov/llama.cpp/discussions/2637)

**Comment by ASH1998** [comment, marked as answer]
Closing the discussion, I figured out a way to do this for server using llama-cpp-python after some tinkering. They too do not support this at this point of time. If anyone is interested in working on this, please do, it will help a lot.

Thanks @ianscrivener for your time, I went on a tangent to discover grammar and it was amazing (I didn't know about this functionality)
